tests/test_logging.py

```
@@ -177,5 +220,135 @@ def test_log_events(caplog, capfd):
         ), f"Expected '{expected_msg}' not found in stderr output"
 
 
-def test_rule_failure(caplog, capfd):
-    run(dpath("logging/test_rule_failure"), check_results=False)
+def test_rule_failure(
+    caplog: pytest.LogCaptureFixture, capfd: pytest.CaptureFixture[str]
```

Ruff (0.14.6)

Unused function argument: `capfd`

(ARG001)